src/lib.rs

```diff
@@ -839,18 +839,19 @@ impl<T> WeakSender<T> {
         if self.channel.queue.is_closed() {
             None
         } else {
-            let old_count = self.channel.sender_count.fetch_add(1, Ordering::Relaxed);
-            if old_count == 0 {
-                // Channel was closed while we were incrementing the count.
-                self.channel.sender_count.store(0, Ordering::Release);
-                None
-            } else if old_count > usize::MAX / 2 {
-                // Make sure the count never overflows, even if lots of sender clones are leaked.
-                process::abort();
-            } else {
-                Some(Sender {
+            match self.channel.sender_count.fetch_update(
+                Ordering::Relaxed,
```

Comment:
The set ordering here should be `Release`.

Reply:
Could you elaborate on what this would guard against? I'm not seeing any other writes that this would need to be ordered with, but I'm probably just missing something